File: xiaohao/数组/14/14.py

def longPrefix(strTest):
    if len(strTest) < 1:
        print("error")
        return ""
    prefix = strTest[0]
    for i in strTest:
        if len(prefix) == 0:
            print("cant't find")
            return ''
        while i.find(prefix) != 0: # 只有当前设定的prefix不是前缀，才进入循环
            prefix = prefix[:-1]
            print(i,prefix)
        '''
        这样的问题是，flow,owfl 结果都是true，不能判断是前缀
        需要使用find方法，find可以返回子字符串出现的位置，为0就是前缀
        if prefix not in i:
            prefix = prefix[:-1]
        '''    
    return prefix


if __name__ == '__main__':
    str0 = ["china","change","choke","chance","child"]
    print(longPrefix(str0))

File: xiaohao/数组/14/14.go
package main

import (
	"fmt"
	"strings"
)

func longPrefix(str []string) string {
	fmt.Println(str)
	if len(str) < 1 {
		fmt.Println("Error")
		return ""
	}

	prefix := str[0]

	for _, k := range str {
		for strings.Index(k, prefix) != 0 { //在k里找prefix
			if len(prefix) == 0 {
				fmt.Println("Can't find")
				return ""
			}
			prefix = prefix[:len(prefix)-1]
		}
	}

	return prefix
}

func main() {
	string0 := []string{"flower", "flow", "floght"}
	fmt.Println(longPrefix(string0))
}
